jwql/instrument_monitors/common_monitors/bias_monitor.py

```diff
@@ -0,0 +1,554 @@
+                                 .format(len(new_files), instrument, aperture))
+                    monitor_run = False
+
+                # Update the query history
+                new_entry = {'instrument': instrument,
+                             'aperture': aperture,
+                             'start_time_mjd': self.query_start,
+                             'end_time_mjd': self.query_end,
+                             'entries_found': len(new_entries),
+                             'files_found': len(new_files),
+                             'run_monitor': monitor_run,
+                             'entry_date': datetime.datetime.now()}
+                self.query_table.__table__.insert().execute(new_entry)
+                logging.info('\tUpdated the query history table')
+
+        logging.info('Bias Monitor completed successfully.')
+
+    def run_pipeline(self, filename, odd_even_rows=False, odd_even_columns=True, 
+                     use_side_ref_pixels=True, group_scale=False):
+        """Runs the early steps of the jwst pipeline (dq_init, saturation, 
+        superbias, refpix) on uncalibrated files and outputs the result.
+        
+        Parameters
+        ----------
+        filename : str
+            File on which to run the pipeline steps
+
+        odd_even_rows : bool
+            Option to treat odd and even rows separately during refpix step
+
+        odd_even_columns : bools
+            Option to treat odd and even columns separately during refpix step
+
+        use_side_ref_pixels : bool
+            Option to perform the side refpix correction during refpix step
+
+        group_scale : bool
+            Option to rescale pixel values to correct for instances where 
+            on-board frame averaging did not result in the proper values
+
+        Returns
+        -------
+        output_filename : str
+            The full path to the calibrated file
+        """
+
+        output_filename = filename.replace('_uncal', '')\
+                                  .replace('.fits', '_superbias_refpix.fits')
+        
+        if not os.path.isfile(output_filename):                       
+            # Run the group_scale and dq_init steps on the input file
+            if group_scale:
+                model = GroupScaleStep.call(filename)
+                model = DQInitStep.call(model)
+            else:
+                model = DQInitStep.call(filename)
+            
+            # Run the saturation and superbias steps
+            model = SaturationStep.call(model)
+            model = SuperBiasStep.call(model)
+
+            # Run the refpix step and save the output
+            model = RefPixStep.call(model, odd_even_rows=odd_even_rows, 
+                odd_even_columns=odd_even_columns, use_side_ref_pixels=use_side_ref_pixels)
+            model.save(output_filename)
+        else:
+            logging.info('\t{} already exists'.format(output_filename))
+            pass
```

Comment:
I don't think this `pass` statement is needed here?  I don't think including it has any notable affect.

Reply:
fixed